pkg/connector/msgconv.go

```diff
@@ -183,12 +186,31 @@ func (tc *TwitterClient) twitterAttachmentToMatrix(ctx context.Context, portal *
 		content.Body += ext
 	}
 
-	content.URL, content.File, err = intent.UploadMediaStream(ctx, portal.MXID, fileResp.ContentLength, true, func(file io.Writer) (*bridgev2.FileStreamResult, error) {
+	audioOnly := attachment.Video != nil && attachment.Video.AudioOnly
+	content.URL, content.File, err = intent.UploadMediaStream(ctx, portal.MXID, fileResp.ContentLength, audioOnly, func(file io.Writer) (*bridgev2.FileStreamResult, error) {
 		n, err := io.Copy(file, fileResp.Body)
 		if err != nil {
 			return nil, err
 		}
-		content.Info.Size = int(n)
+		if audioOnly && ffmpeg.Supported() {
+			outFile, err := ffmpeg.ConvertPath(ctx, file.(*os.File).Name(), ".ogg", []string{}, []string{"-c:a", "libopus"}, false)
+			if err == nil {
+				mimeType = "audio/ogg"
+				content.Info.MimeType = mimeType
+				content.Info.Width = 0
+				content.Info.Height = 0
+				content.MsgType = bridgeEvt.MsgAudio
+				return &bridgev2.FileStreamResult{
+					ReplacementFile: outFile,
+					MimeType:        mimeType,
+					FileName:        content.Body + ".ogg",
```

Comment:
`content.Body` should be updated to have `.ogg` too, this returned file name is only for the media upload request and not the event content itself